CRM/Contact/StateMachine/Search.php

```diff
@@ -103,15 +103,10 @@ public function taskName($controller, $formName = 'Search') {
     }
     $this->_controller->set('task', $value);
 
-    if ($value) {
```

Comment:
Any idea how to get to this line without $value to test?

Reply:
This was causing a bug in some situations, as it was falling back to the contact task and would try and return the default task for contact tasks no matter what context we were in (eg. in member context).  I think you could trigger it by editing a smart search and changing the "View results as" to be something other than contacts.  But it should not be possible to get there with no value for $value unless you are doing something wierd and it doesn't make sense (in my opinion) to fall back to the contact task handler, as it defines a different set of tasks which are potentially not applicable to the current context.

By removing the check for $value we are always calling the relevant getTask() handler for the context.